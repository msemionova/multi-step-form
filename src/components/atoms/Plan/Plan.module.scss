.plan {
  width: 100%;
  display: inline-block;
  background-color: var(--color-white);
  border-radius: var(--border-radius-medium);
  border: var(--border-secondary);
  transition: background-color, border 200ms;
  padding: 2rem 1.6rem;
  margin-bottom: 3.2rem;

  .icon {
    width: 4rem;
    height: 4rem;
    margin-bottom: 4.4rem;

    &--arcade {
      background: no-repeat center/100% url("~/public/assets/images/icon-arcade.svg");
    }

    &--advanced {
      background: no-repeat center/100% url("~/public/assets/images/icon-advanced.svg");
    }

    &--pro {
      background: no-repeat center/100% url("~/public/assets/images/icon-pro.svg");
    }
  }

  h3 {
    font: var(--typography-input-text);
    text-transform: capitalize;
  }

  span {
    font: var(--typography-input-label);
    color: var(--color-gray-text-light);
  }

  p {
    font: var(--typography-small-text);
  }
}

input[type="radio"] {
  display: none;
}

input[type="radio"]:checked + .plan {
  background-color: var(--color-gray-background);
  border: var(--border-primary);
}
