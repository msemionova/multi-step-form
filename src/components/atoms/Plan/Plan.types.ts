export type PlanProps = {
  id: string,
  priceMonthly: number,
  priceYearly: number,
  type?: 'monthly' | 'yearly',
  currency?: string,
  isChecked?: boolean
}
