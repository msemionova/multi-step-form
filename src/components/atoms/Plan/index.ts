export { default as Plan } from './Plan';
export type { PlanProps } from './Plan.types';
