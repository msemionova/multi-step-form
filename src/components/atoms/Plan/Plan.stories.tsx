import type { Meta, StoryObj } from '@storybook/react';

import Plan from './Plan';

const meta: Meta<typeof Plan> = {
  component: Plan,
};

export default meta;
type Story = StoryObj<typeof Plan>;

export const ArcadeActive: Story = {
  args: {
    id: 'arcade',
    priceMonthly: 5,
    priceYearly: 50,
    isChecked: true
  }
};

export const Advanced: Story = {
  args: {
    id: 'advanced',
    priceMonthly: 7,
    priceYearly: 70
  }
};

export const Pro: Story = {
  args: {
    id: 'pro',
    priceMonthly: 99,
    priceYearly: 999
  }
};
