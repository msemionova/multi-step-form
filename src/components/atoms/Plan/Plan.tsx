import styles from './Plan.module.scss';
import { PlanProps } from './Plan.types';

const Plan = ({
  id,
  priceMonthly,
  priceYearly,
  type = 'monthly',
  currency = '$',
  isChecked = false
}: PlanProps) => {
  const price: string =
    type === 'monthly' ? `${String(priceMonthly)}/mo` : `${String(priceYearly)}/yr`;
  const iconClasses: string = [styles.icon, styles[`icon--${id}`]].filter(Boolean).join(' ');

  return (
    <div>
      <input type='radio' name='plans' id={id} value={id} defaultChecked={isChecked} />
      <label className={styles.plan} htmlFor={id}>
        <i className={iconClasses} />
        <h3>{id}</h3>
        <span>{`${currency}${price}`}</span>
        {type === 'yearly' && <p>2 monts free</p>}
      </label>
    </div>
  );
};

export default Plan;
