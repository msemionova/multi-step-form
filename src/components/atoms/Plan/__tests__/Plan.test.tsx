import Plan from '../Plan';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

describe('<Plan />', () => {
  it('renders correctly without required nor label...', () => {
    const tree = renderer
      .create(<Plan id='arcade' priceMonthly={9} priceYearly={999} isChecked={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
