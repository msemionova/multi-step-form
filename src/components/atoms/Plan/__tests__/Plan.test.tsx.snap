// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Plan /> renders correctly without required nor label... 1`] = `
<div>
  <input
    defaultChecked={true}
    id="plan"
    name="plans"
    type="radio"
    value="plan"
  />
  <label
    className="plan"
    htmlFor="plan"
  >
    <div
      className="icon icon--plan"
    />
    <h3>
      plan
    </h3>
    <span>
      $9/mo
    </span>
  </label>
</div>
`;
